src/Tests/Libs/GirTest-0.1.Tests/StringTest.cs

```diff
@@ -0,0 +1,62 @@
+﻿using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace GirTest.Tests;
+
+[TestClass, TestCategory("BindingTest")]
+public class StringTest : Test
+{
+    // This string is 4 bytes in the glib filename encoding (ISO-8859-1) that is
+    // used for these tests, but is 5 bytes in UTF-8.
+    private const string TestString = "a\u00f6bc";
+
+    [TestMethod]
+    public void InUtf8ParameterShouldSucceed()
+    {
+        StringTester.Utf8In(TestString).Should().Be(TestString.Length);
+
+        StringTester.Utf8InNullable(TestString).Should().Be(TestString.Length);
+        StringTester.Utf8InNullable(null).Should().Be(-1);
+    }
+
+    [TestMethod]
+    public void InFilenameParameterShouldSucceed()
+    {
+        StringTester.FilenameIn(TestString).Should().Be(TestString.Length);
+
+        StringTester.FilenameInNullable(TestString).Should().Be(TestString.Length);
+        StringTester.FilenameInNullable(null).Should().Be(-1);
+    }
+
+    [TestMethod]
+    public void ReturnUtf8ParameterShouldSucceed()
+    {
+        StringTester.Utf8ReturnTransferFull(TestString).Should().Be(TestString);
+        StringTester.Utf8ReturnTransferNone(TestString).Should().Be(TestString);
+
+        StringTester.Utf8ReturnNullableTransferFull(TestString).Should().Be(TestString);
+        StringTester.Utf8ReturnNullableTransferFull(null).Should().Be(null);
+
+        StringTester.Utf8ReturnNullableTransferNone(TestString).Should().Be(TestString);
+        StringTester.Utf8ReturnNullableTransferNone(null).Should().Be(null);
+    }
+
+    [TestMethod]
+    public void ReturnFilenameParameterShouldSucceed()
+    {
+        StringTester.FilenameReturnTransferFull(TestString).Should().Be(TestString);
+        StringTester.FilenameReturnTransferNone(TestString).Should().Be(TestString);
+
+        StringTester.FilenameReturnNullableTransferFull(TestString).Should().Be(TestString);
+        StringTester.FilenameReturnNullableTransferFull(null).Should().Be(null);
+
+        StringTester.FilenameReturnNullableTransferNone(TestString).Should().Be(TestString);
```

Comment:
Shouldn't be the returned string different to the input string as it should get converted to regular UTF8?

Reply:
It should be identical here as long as the managed string's contents can be represented in the filename encoding (`ISO-8859-1` in these unit tests). 
- The input parameter is also a filename type, so the managed (UTF-16) string is converted to the filename encoding via `PlatformStringOwnedHandle.Create()` before being passed to the native function
- The return type is a `PlatformStringUnownedHandle` (filename encoding) which Is then converted back to a managed UTF-16 string